main/src/ca/uwaterloo/flix/language/ast/AtomicOp.scala

```diff
@@ -62,7 +64,7 @@ object AtomicOp {
 
   case object ArrayLength extends AtomicOp
 
-  case class StructNew(sym: Symbol.StructSym, fields: List[Symbol.StructFieldSym], pure: Boolean) extends AtomicOp
+  case class StructNew(sym: Symbol.StructSym, fields: List[Symbol.StructFieldSym], pure: Mutable) extends AtomicOp
```

Comment:
pure -> mutability

I also think we should consider putting this information about between the symbol (which is always first) and before the fields.